## Target Variable Analysis

Target variable analysis is one of the first steps in data understanding.  
It helps us understand the nature and distribution of the variable we want to predict.

- In **classification**, we check how classes are distributed.
- In **regression**, we study statistical properties like mean, spread, and skewness.

This step is important because:
- It helps detect class imbalance.
- It gives an idea about data distribution.
- It guides model selection and evaluation strategy.
---

### Dataset Used

For this analysis, we use the **tips dataset** available from the Seaborn library.

In [1]:
import seaborn as sns
df = sns.load_dataset('tips')

---

### Target Variable Count 

In [2]:
df["tip"].value_counts()

tip
2.00    33
3.00    23
4.00    12
2.50    10
5.00    10
        ..
1.47     1
1.17     1
4.67     1
5.92     1
1.75     1
Name: count, Length: 123, dtype: int64

**Explanation:**
- This shows how many times each tip value appears in the dataset.
- Tip = 2.00 appears 33 times (most common).
- Tip = 3.00 appears 23 times.
- Some tip values appear only once (rare cases).
- Total unique tip values = 123.

**This shows:**
- `value_counts()` counts frequency of each unique value in a column.
- Helps to understand:
  - Most frequent values
  - Rare values
  - Distribution of data
- Useful for checking imbalance and variability in target variable.

### Target Variable Distribution (Percentage)

In [3]:
df["tip"].value_counts(normalize=True)

tip
2.00    0.135246
3.00    0.094262
4.00    0.049180
2.50    0.040984
5.00    0.040984
          ...   
1.47    0.004098
1.17    0.004098
4.67    0.004098
5.92    0.004098
1.75    0.004098
Name: proportion, Length: 123, dtype: float64

**Explanation :**
- This shows the proportion (percentage) of each tip value.
- Tip = 2.00 makes up about 13.5% of total data (most common).
- Tip = 3.00 makes up about 9.4%.
- Rare values have very small proportions (~0.4%).
- Total of all values = 1 (100%).

**This shows:**
- `value_counts(normalize=True)` gives relative frequency instead of count.
- Helps to understand:
  - Percentage distribution of values
  - Dominant vs rare values
- Useful for comparing importance of values and detecting imbalance.

## Statastical summary for target variable

In [4]:
df["tip"].describe()

count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64

**Explanation:**
- Total values = 244.
- Average (mean) tip ≈ 3.00.
- Standard deviation ≈ 1.38 → moderate spread.
- Minimum tip = 1.00, Maximum tip = 10.00.
- 25% values ≤ 2.00 (lower quartile).
- 50% values ≤ 2.90 (median).
- 75% values ≤ 3.56 (upper quartile).

**This shows:**
- `describe()` gives statistical summary of numerical data.
- Includes:
  - Count, Mean, Standard Deviation
  - Min, Max
  - Quartiles (25%, 50%, 75%)
- Helps to understand:
  - Central tendency (mean, median)
  - Spread of data
  - Range and possible outliers

## checking skewness in target variable.

In [5]:
df["tip"].skew()

np.float64(1.4654510370979401)

**Explanation:**
- Skewness ≈ 1.47 (positive value).
- Data is right-skewed (long tail on the right side).
- Most tip values are smaller, with few large values.

**This shows:**
- Skewness measures asymmetry of data distribution.
- If:
  - Skew = 0 → symmetric data
  - Skew > 0 → right-skewed (more small values, few large)
  - Skew < 0 → left-skewed (more large values, few small)
- Helps to understand distribution shape and need for transformation.